edb/schema/scalars.py

```diff
@@ -354,10 +359,50 @@ def apply(
         self.scls = scls
         assert isinstance(scls, ScalarType)
 
-        enum_values = scls.get_enum_values(schema)
-        if enum_values:
-            raise errors.UnsupportedFeatureError(
-                f'altering enum composition is not supported')
+        cur_labels = scls.get_enum_values(schema)
+
+        if cur_labels:
+
+            if self.removed_bases and not self.added_bases:
+                raise errors.SchemaError(
+                    f'cannot DROP EXTENDING enum')
+
+            all_bases = []
+
+            for bases, pos in self.added_bases:
+                for base in bases:
+                    if isinstance(base, AnonymousEnumTypeShell):
+                        is_enum_base = True
+                    elif isinstance(base, s_types.TypeShell):
+                        is_enum_base = base.resolve(schema).is_enum(schema)
+                    else:
+                        is_enum_base = base.is_enum(schema)
```

Comment:
This code seems wrong. If you have two new bases "Enum, NonEnum" the `is_enum_base` would be `False` after the loop.

Reply:
The `is_enum_base` is only used inside the loop to weed out the cases where there's something other than enums in the bases list. Beyond the loop it is assumed that the bases are enums-only, since otherwise the loop would have produced an error.

Reply:
my bad, you're right